File: app/src/main/java/com/nirbhor/app/ui/components/Scaffold.kt
package com.nirbhor.app.ui.components

import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.layout.windowInsetsPadding
import androidx.compose.foundation.layout.WindowInsets
import androidx.compose.foundation.layout.statusBars
import androidx.compose.foundation.layout.navigationBars
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.automirrored.filled.ArrowBack
import androidx.compose.material3.Icon
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.vector.ImageVector
import androidx.compose.ui.unit.dp
import com.nirbhor.app.ui.theme.Dimens
import com.nirbhor.app.ui.theme.NirbhorTheme

/** Header bar for pushed screens: back chevron + title on a card surface with a hairline bottom border. */
@Composable
fun NirbhorTopBar(
    title: String,
    modifier: Modifier = Modifier,
    onBack: (() -> Unit)? = null,
    trailing: (@Composable () -> Unit)? = null,
) {
    val colors = NirbhorTheme.colors
    Column(modifier = modifier.fillMaxWidth().background(colors.card)) {
        Row(
            modifier = Modifier
                .fillMaxWidth()
                .windowInsetsPadding(WindowInsets.statusBars)
                .padding(horizontal = Dimens.denseHeaderPadding, vertical = 12.dp),
            verticalAlignment = Alignment.CenterVertically,
        ) {
            if (onBack != null) {
                Box(
                    Modifier.size(Dimens.tapMin).clip(CircleShape).clickable(onClick = onBack),
                    contentAlignment = Alignment.Center,
                ) {
                    Icon(Icons.AutoMirrored.Filled.ArrowBack, contentDescription = "Back", tint = colors.ink)
                }
                Spacer(Modifier.width(6.dp))
            }
            Text(title, style = NirbhorTheme.type.header, color = colors.ink, modifier = Modifier.weight(1f))
            if (trailing != null) trailing()
        }
        Box(Modifier.fillMaxWidth().height(1.dp).background(colors.line))
    }
}

/** One bottom-nav tab. */
data class NavTab(
    val route: String,
    val icon: ImageVector,
    val labelBn: String,
    val labelEn: String,
)

/** Four-tab bottom nav. Active item gets calm text on a calm-soft radius-12 pill. */
@Composable
fun BottomNavBar(
    tabs: List<NavTab>,
    currentRoute: String?,
    isBangla: Boolean,
    onSelect: (NavTab) -> Unit,
    modifier: Modifier = Modifier,
) {
    val colors = NirbhorTheme.colors
    Column(modifier = modifier.fillMaxWidth().background(colors.card)) {
        Box(Modifier.fillMaxWidth().height(1.dp).background(colors.line))
        Row(
            modifier = Modifier
                .fillMaxWidth()
                .windowInsetsPadding(WindowInsets.navigationBars)
                .padding(horizontal = 8.dp, vertical = 6.dp),
            horizontalArrangement = Arrangement.SpaceBetween,
        ) {
            tabs.forEach { tab ->
                val active = currentRoute == tab.route
                Column(
                    modifier = Modifier
                        .weight(1f)
                        .height(Dimens.bottomNavItem)
                        .clip(RoundedCornerShape(12.dp))
                        .clickable { onSelect(tab) },
                    horizontalAlignment = Alignment.CenterHorizontally,
                    verticalArrangement = Arrangement.Center,
                ) {
                    Box(
                        Modifier
                            .clip(RoundedCornerShape(12.dp))
                            .background(if (active) colors.calmSoft else androidx.compose.ui.graphics.Color.Transparent)
                            .padding(horizontal = 16.dp, vertical = 4.dp),
                    ) {
                        Icon(
                            tab.icon,
                            contentDescription = if (isBangla) tab.labelBn else tab.labelEn,
                            tint = if (active) colors.calm else colors.ink3,
                            modifier = Modifier.size(Dimens.navIcon),
                        )
                    }
                    Spacer(Modifier.height(2.dp))
                    Text(
                        text = if (isBangla) tab.labelBn else tab.labelEn,
                        style = NirbhorTheme.type.statusPill,
                        color = if (active) colors.calm else colors.ink3,
                    )
                }
            }
        }
    }
}

File: lib/ui/components/scaffold.dart
import 'package:flutter/material.dart';

import '../../theme/theme.dart';

/// Header bar for pushed screens: back chevron + title on a card surface with a hairline bottom
/// border.
class NirbhorTopBar extends StatelessWidget {
  const NirbhorTopBar({
    super.key,
    required this.title,
    this.onBack,
    this.trailing,
  });

  final String title;
  final VoidCallback? onBack;
  final Widget? trailing;

  @override
  Widget build(BuildContext context) {
    final colors = context.colors;
    return ColoredBox(
      color: colors.card,
      child: Column(
        children: [
          SafeArea(
            bottom: false,
            child: Padding(
              padding: const EdgeInsets.symmetric(
                horizontal: Dimens.denseHeaderPadding,
                vertical: 12,
              ),
              child: Row(
                children: [
                  if (onBack != null) ...[
                    Semantics(
                      button: true,
                      label: context.tr('পিছনে যান', 'Back'),
                      child: InkWell(
                        customBorder: const CircleBorder(),
                        onTap: onBack,
                        child: SizedBox(
                          width: Dimens.tapMin,
                          height: Dimens.tapMin,
                          child: Icon(Icons.arrow_back, color: colors.ink),
                        ),
                      ),
                    ),
                    const SizedBox(width: 6),
                  ],
                  Expanded(
                    child: Text(
                      title,
                      style: context.type.header.copyWith(color: colors.ink),
                      maxLines: 1,
                      overflow: TextOverflow.ellipsis,
                    ),
                  ),
                  ?trailing,
                ],
              ),
            ),
          ),
          Container(height: 1, color: colors.line),
        ],
      ),
    );
  }
}

/// One bottom-nav tab.
class NavTab {
  const NavTab(this.route, this.icon, this.labelBn, this.labelEn);

  final String route;
  final IconData icon;
  final String labelBn;
  final String labelEn;
}

/// Four-tab bottom nav. Active item gets calm text on a calm-soft radius-12 pill.
class BottomNavBar extends StatelessWidget {
  const BottomNavBar({
    super.key,
    required this.tabs,
    required this.currentRoute,
    required this.onSelect,
  });

  final List<NavTab> tabs;
  final String? currentRoute;
  final ValueChanged<NavTab> onSelect;

  @override
  Widget build(BuildContext context) {
    final colors = context.colors;
    return ColoredBox(
      color: colors.card,
      child: Column(
        mainAxisSize: MainAxisSize.min,
        children: [
          Container(height: 1, color: colors.line),
          SafeArea(
            top: false,
            child: Padding(
              padding: const EdgeInsets.symmetric(horizontal: 8, vertical: 6),
              child: Row(
                children: [
                  for (final tab in tabs)
                    Expanded(
                      child: _NavItem(
                        tab: tab,
                        active: currentRoute == tab.route,
                        onTap: () => onSelect(tab),
                      ),
                    ),
                ],
              ),
            ),
          ),
        ],
      ),
    );
  }
}

class _NavItem extends StatelessWidget {
  const _NavItem({required this.tab, required this.active, required this.onTap});

  final NavTab tab;
  final bool active;
  final VoidCallback onTap;

  @override
  Widget build(BuildContext context) {
    final colors = context.colors;
    return Semantics(
      selected: active,
      button: true,
      child: InkWell(
        borderRadius: BorderRadius.circular(12),
        onTap: onTap,
        child: SizedBox(
          height: Dimens.bottomNavItem,
          child: Column(
            mainAxisAlignment: MainAxisAlignment.center,
            children: [
              Container(
                decoration: BoxDecoration(
                  color: active ? colors.calmSoft : Colors.transparent,
                  borderRadius: BorderRadius.circular(12),
                ),
                padding: const EdgeInsets.symmetric(horizontal: 16, vertical: 4),
                child: Icon(
                  tab.icon,
                  size: Dimens.navIcon,
                  color: active ? colors.calm : colors.ink3,
                ),
              ),
              const SizedBox(height: 2),
              Text(
                context.isBangla ? tab.labelBn : tab.labelEn,
                style: context.type.statusPill.copyWith(
                  color: active ? colors.calm : colors.ink3,
                ),
              ),
            ],
          ),
        ),
      ),
    );
  }
}
